assemble-and-restart

```diff
@@ -2,6 +2,51 @@
 set -x
 set -eo pipefail
 
+# If WorkingDir is injected as an env from odo and destination path is not equal to WorkingDir
+if [ ! -z "${ODO_S2I_WORKING_DIR}" ] && ([ "${ODO_S2I_SRC_BIN_PATH}" != "${ODO_S2I_WORKING_DIR}" ] || [ "${ODO_S2I_DEPLOYMENT_DIR}" != "${ODO_S2I_WORKING_DIR}" ]); then
+
+    # If ODO_SRC_BACKUP_DIR is injected by odo
+    if [ -n  "$ODO_SRC_BACKUP_DIR" ]; then
+
+        # If it doesn't exit create it
+        if [ ! -d ${ODO_SRC_BACKUP_DIR} ]; then
+            mkdir -p ${ODO_SRC_BACKUP_DIR}/src
+        fi
+
+        # Backup the sources in DestinationDir to ODO_SRC_BACKUP_DIR because the assemble script for some s2i images
+        # moves sources from DestinationDir to WorkingDir thereby deleting it there(DestinationDir). So, back it up
+        # for partial push use cases like watch
+        rsync -r ${ODO_S2I_SRC_BIN_PATH}/src/. ${ODO_SRC_BACKUP_DIR}/src/
```

Comment:
it would be better to use `rsync -a` to make sure that it is really identical copy

Reply:
Done